vscode/webviews/mentions/mentionMenu/MentionMenu.tsx

```diff
@@ -174,20 +174,36 @@ export const MentionMenu: FunctionComponent<
                     updateMentionMenuParams({
                         parentItem: {
                             id: RemoteFileProvider.providerUri,
-                            title: 'Sourcegraph Files',
-                            queryLabel: 'Enter file path to search.',
-                            emptyLabel: `No files found in ${openCtxItem.mention.data.repoName} repository`,
+                            title: 'Remote Files',
+                            queryLabel: 'Enter file path to search',
+                            emptyLabel: `No files found in ${openCtxItem?.mention?.data.repoName} repository`,
                         },
                     })
-                    setEditorQuery(() => `@${openCtxItem.mention?.data?.repoName}:`)
+
+                    setEditorQuery(currentText => {
+                        const replaceText = `@${mentionQuery.text}`
+                        const mentionStartIndex = currentText.indexOf(replaceText)
+
+                        if (mentionStartIndex !== -1) {
+                            const mentionEndIndex = mentionStartIndex + replaceText.length
+                            return (
+                                currentText.slice(0, mentionStartIndex) +
+                                `@${openCtxItem.mention?.data?.repoName}:` +
+                                currentText.slice(mentionEndIndex)
+                            )
+                        }
+
+                        return currentText
```

Comment:
This feels a little fragile, but good enough for now. I believe quinn has made it possible to get hold of the lexical state, but don't know enough to guide you here without diving in super deep. Also if I am not mistaken you can replace this with

```suggestion
                        return currentText.replace(replaceText, `@${openCtxItem.mention?.data?.repoName}:`)
```

Reply:
Lol, I absolutely did the same, but then I don't know why I copied the implementation from the `onProviderSelection` method above.

Let me update it.